[3007.x] Merge 3006.x into 3007.x

s0undt3ch opened the pull request:
### What does this PR do?
See title

garethgreenaway committed:
increase test coverage for textfsm module.

garethgreenaway committed:
fixing lint issues

garethgreenaway committed:
Update tests/pytests/unit/modules/test_textfsm_mod.py

Co-authored-by: Pedro Algarvio <pedro@algarvio.me>

garethgreenaway committed:
Update tests/pytests/unit/modules/test_textfsm_mod.py

Co-authored-by: Pedro Algarvio <pedro@algarvio.me>

garethgreenaway committed:
Update tests/pytests/unit/modules/test_textfsm_mod.py

Co-authored-by: Pedro Algarvio <pedro@algarvio.me>

garethgreenaway committed:
Update tests/pytests/unit/modules/test_textfsm_mod.py

Co-authored-by: Pedro Algarvio <pedro@algarvio.me>

garethgreenaway committed:
Running pre-commit manually

garethgreenaway committed:
renaming virtual test name, fixing nesting.

garethgreenaway committed:
removing parens from pytest.fixture.

dead10ck committed:
delete s3fs: delete files from cache when deleted from bucket

s0undt3ch committed:
Remove custom fixture scope ordering code

Signed-off-by: Pedro Algarvio <palgarvio@vmware.com>

twangboy committed:
Fix test_log_beacon.py::test_log_match test on Windows

terminalmage committed:
Fix file.directory clobbering backupname path with test=True

terminalmage committed:
Add test case

terminalmage committed:
Add changelog entry

twangboy committed:
Fix an issue with the win_pki execution module

Change the _cmd_run function so that it raises a CommandExecutionError
when the command fails so that it doesn't try to parse the output of the
command as JSON
Also raise a CommandExecutionError when it recieves invalid JSON

dwoz committed:
Fix content-type backwards compatability

s0undt3ch committed:
Fix lint and formatting issues

s0undt3ch committed:
Add Fedora 39 support to create repo

twangboy committed:
Make the win_function.get_sam_name function handle Everyone

s0undt3ch committed:
Add `textfsm` to our requirements so that tests depending on it actually run

twangboy committed:
Honor saltenv in refresh_db on Windows

s0undt3ch committed:
Bump to `pytest-timeout==2.3.1`

s0undt3ch committed:
Bump the timeout for the pyobjects unit tests

twangboy committed:
Fix issue with refresh_db on Windows

When a new package definition file was added to the repo you had to run
refresh_db twice to get it to show up in the package databse. This
clears the cache before refreshing the database to force changes.

twangboy committed:
Remove get_package_info from missing_examples

twangboy committed:
Upgrade WIX to 3.14

terminalmage committed:
black

twangboy committed:
Fix test on Windows

twangboy committed:
Whitelist package tests for Windows

s0undt3ch committed:
Upgrade to Pytest 8.1.x

twangboy committed:
Fix package tests

s0undt3ch committed:
Daylight savings suck!

twangboy committed:
Remove offending colon

twangboy committed:
Don't use curl on Windows

Commit:
move fix from #66164 for $65703

s0undt3ch committed:
Upgrade to `flaky==3.8.1`

s0undt3ch committed:
Increase test timeout

Commit:
attempt #1 to fix tests

dwoz committed:
Check for content-type after headers dict has been populated

Commit:
stupid mistake correction

s0undt3ch committed:
Bump to `pygments==2.17.2` due to https://github.com/advisories/GHSA-mrwq-x4v8-fh7p

s0undt3ch committed:
Merge 3006.x into 3007.x

s0undt3ch committed:
Merge 3006.x into 3007.x

s0undt3ch committed:
Merge 3006.x into 3007.x

twangboy committed:
Make the tests run on Windows

s0undt3ch committed:
Bump timeout

s0undt3ch committed:
Minor timeout adjustments

s0undt3ch committed:
Randomize names and prefer `salt_master.state_tree.base`

twangboy committed:
Skip state test on Windows

twangboy committed:
Revert weird change

s0undt3ch committed:
Increase timeouts

s0undt3ch committed:
Skip problematic MacOS test in specific conditions

s0undt3ch committed:
Merge 3006.x into 3007.x

s0undt3ch committed:
Add our own GH cache action

This actions wraps actions/cache and forces an early exit if 'fail-on-cache-miss' is true

Based on the work from https://github.com/saltstack/salt/pull/66240

s0undt3ch committed:
Remove invalid properties from existing GH actions

s0undt3ch committed:
Force cache invalidation by setting cache seed to 1

s0undt3ch committed:
Get AWS regions and SPB_ENVIRONMENT from metadata endpoint

Commit:
Initial commit of s3 caching

twangboy committed:
Fix issue with win_user.add

win_useradd.add now allows you to add new users that have only integers in
the username.
It also adds tests for the win_useradd module.

s0undt3ch committed:
Merge branch '3007.x' of github.com:saltstack/salt into hotfix/merge-forward-into-3007.x

s0undt3ch committed:
Merge 3006.x into 3007.x

s0undt3ch committed:
Remove log call which no longer makes sense

s0undt3ch committed:
Adjust some tests to get passing nightly builds again

twangboy committed:
Move test_win_useradd.py to functional

twangboy committed:
Fix some test issues, use sort

twangboy committed:
Don't use configure_loader_modules

twangboy committed:
Use sorted instead of sort

twangboy committed:
Remove vcredist and ucrt from Windows install

twangboy committed:
Fix failing package tests

s0undt3ch committed:
Upgrade to ``pytest-salt-factories==1.0.0``

s0undt3ch committed:
Upgrade to ``pytest-salt-factories==1.0.0``

s0undt3ch committed:
Bump functional test splits to 4

s0undt3ch committed:
Bump functional test splits to 4

s0undt3ch committed:
Stop ignoring the host keys, query them and write them down as accepted.

Akm0d committed:
Fix: Prevent full system upgrade in pkg.installed for Arch Linux

This change modifies the logic in the pacman module to only append the '-u' flag to the pacman command if sysupgrade is explicitly set to True. This prevents the pkg.installed state from triggering a full system upgrade by default on Arch Linux systems.

s0undt3ch committed:
Upgrade to ``pytest-salt-factories==1.0.1``

s0undt3ch committed:
Reap child processes after each of the ssh tests in case of timeouts

ScriptAutomate committed:
Reduce test range in PRs and branch merges

Commit:
Fix RedHat family systemd restart on upgrade, and updated tests

s0undt3ch committed:
Allow choosing which OS to run tests against using labels

s0undt3ch committed:
Add tools command to sync known OS'es with GH test labels

s0undt3ch committed:
Mandatory OS slugs are now defined in ``cicd/shared-gh-workflows-context.yml``

s0undt3ch committed:
Use ``attrs`` to simplify OS definitions

s0undt3ch committed:
Bump to ``dorny/paths-filter@v3``

Commit:
Ensure path for systemctl to stop spoofing (security fix moved from classic spec) and debug helper scripts

Commit:
Ensure symbolic links are created before using systemctl

s0undt3ch committed:
Just "grep" what we're after

Signed-off-by: Pedro Algarvio <palgarvio@vmware.com>

s0undt3ch committed:
Always print out the list of jobs to run

s0undt3ch committed:
Always print out the defined testrun

s0undt3ch committed:
Always print the collected runners

s0undt3ch committed:
Make jinja rendering strict. Undefined variable use throws error.

s0undt3ch committed:
Create `get_cicd_shared_context()` and `get_golden_images()`

Commit:
Add leading slash to salt helper file paths as per dh_links requirement

s0undt3ch committed:
Add what OS'es run tests as a step summary

s0undt3ch committed:
Merge branch '3007.x' of github.com:saltstack/salt into hotfix/merge-forward-into-3007.x

s0undt3ch committed:
Fix cache prefixes

s0undt3ch committed:
Check against an array, not a string

s0undt3ch committed:
Merge 3006.x into 3007.x

s0undt3ch committed:
For now, ignore the host keys

s0undt3ch committed:
Merge branch '3007.x' of github.com:saltstack/salt into hotfix/merge-forward-into-3007.x

nicholasmhughes committed:
fixes saltstack/salt#66284 x509.certificate_managed - ca_server did not return a certificate

twangboy commented on pkg/windows/msi/Product.wxs:
```diff
@@ -267,7 +258,6 @@ IMCAC - Immediate Custom Action - It's immediate
     <!-- Leaving registry keys would mean the product is still installed -->
     <Feature Id="ProductFeature" Title="Minion" Level="1">
       <ComponentGroupRef Id="ProductComponents" />
-      <Feature Id="VC143" Title="VC++ 2022" AllowAdvertise="no" Display="hidden"><MergeRef Id="MSM_VC143_CRT" /></Feature>
```
We need to keep all changes in this file... we want to install the VC143 Merge Module.

twangboy commented on pkg/windows/msi/README-how-to-build.md:
```diff
@@ -10,8 +10,6 @@ You need
 - [Wix 3](http://wixtoolset.org/releases/)<sup>**</sup>
 - [Build tools 2015](https://www.microsoft.com/en-US/download/confirmation.aspx?id=48159)<sup>**</sup>
-- Microsoft_VC143_CRT_x64.msm from Visual Studio 2015<sup>**</sup>
-- Microsoft_VC143_CRT_x86.msm from Visual Studio 2015<sup>**</sup>
```
Don't remove this

twangboy commented on pkg/windows/msi/build_pkg.ps1:
```diff
@@ -608,7 +578,6 @@ Push-Location $SCRIPT_DIR
     -dDisplayVersion="$Version" `
     -dInternalVersion="$INTERNAL_VERSION" `
     -dDISCOVER_INSTALLDIR="$($DISCOVER_INSTALLDIR[$i])" `
-    -dWEBCACHE_DIR="$WEBCACHE_DIR" `
```
We want to keep all this stuff too... don't remove

twangboy commented on pkg/windows/nsis/installer/Salt-Minion-Setup.nsi:
```diff
@@ -524,171 +524,6 @@ InstallDirRegKey HKLM "${PRODUCT_DIR_REGKEY}" ""
-    # This directory gets removed upon completion
-    SetOutPath "$PLUGINSDIR\"
-    File /r "..\..\prereqs\"
-SectionEnd
```
We keep this section

twangboy commented on pkg/windows/nsis/installer/Salt-Minion-Setup.nsi:
```diff
@@ -524,171 +524,6 @@ InstallDirRegKey HKLM "${PRODUCT_DIR_REGKEY}" ""
-    StrCmp $0 "1" +2 0
-    Call InstallVCRedist
-
-SectionEnd
```
We also need to keep this section (`-install_vcredist_2022`)

twangboy commented on pkg/windows/nsis/installer/Salt-Minion-Setup.nsi:
```diff
@@ -524,171 +524,6 @@ InstallDirRegKey HKLM "${PRODUCT_DIR_REGKEY}" ""
-                /SD IDOK
-        ${EndIf}
-
-FunctionEnd
```
This whole function needs to stay (`InstallVCRedist`)

twangboy commented on pkg/windows/nsis/installer/Salt-Minion-Setup.nsi:
```diff
@@ -763,7 +598,6 @@ Function .onInit
         ${EndIf}
     ${EndIf}
 
-    InitPluginsDir
```
We need this line

twangboy commented on pkg/windows/nsis/tests/setup.ps1:
```diff
@@ -70,19 +68,6 @@ $directories | ForEach-Object {
-        exit 1
-    }
-}
-
```
Don't change this file...

twangboy commented on pkg/windows/prep_salt.ps1:
```diff
@@ -183,18 +171,6 @@ $scripts | ForEach-Object {
-    exit 1
-}
-
 #-------------------------------------------------------------------------------
```
Don't change this file

s0undt3ch committed:
We still need VC Redist on 3007.x

twangboy reviewed (changes requested):
In 3006 I removed installing VC Redist 2013
Our 3007 packages require VC Redist 2022, so we don't want to remove that code there.

ScriptAutomate committed:
Replace AlmaLinux with Rocky Linux

nicholasmhughes committed:
use importorskip

s0undt3ch committed:
Windows needs more complex passwords. Use the auto generated one.

Reduce fixture duplication.

s0undt3ch committed:
Do not run nightly or scheduled builds on private repos

s0undt3ch committed:
Define additional `needs` for a few jobs

s0undt3ch committed:
Merge 3006.x into 3007.x

s0undt3ch committed:
Temporarily skip some tests on Rocky Linux 8 Arm64

s0undt3ch committed:
Remove extra logic missed in 68131ce7ab248255feb50436a8b1c6c44813c762

This allowed the tests in https://github.com/saltstack/salt/pull/66169
to pass when they shouldn't

s0undt3ch committed:
Merge branch '3007.x' of github.com:saltstack/salt into hotfix/merge-forward-into-3007.x

s0undt3ch committed:
Merge 3006.x into 3007.x

s0undt3ch committed:
Ignore the hosts key since it's not using the test SSHD server

s0undt3ch committed:
Merge branch '3007.x' of github.com:saltstack/salt into hotfix/merge-forward-into-3007.x
